src/main/java/com/example/bookstoreapp/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.example.bookstoreapp.repository.impl;
+
+import com.example.bookstoreapp.entity.Book;
+import com.example.bookstoreapp.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
+    @PersistenceContext
```

Comment:
why do you need this annotation?

Reply:
Because this annotation forces Spring to inject a fake instance of entityManager, a so-called proxy, which will find the corresponding entityManager for each thread.